### PR TITLE
Snom patch: rtp_echo action

### PR DESCRIPTION
**Pull request by pbertera**
**rtp_echo:** This patch introduces a new action <rtp_echo> allowing to stop and start the RTP echo from the scenario file.

**Example:**
```xml
<nop>
        <action>
                <rtp_echo value="0"/> <!-- stop the RTP echo -->
        </action>
</nop>
<pause milliseconds="15000"/>
<nop>
        <action>
                <rtp_echo value="1"/> <!-- start the RTP echo -->
        </action>
</nop>
```
*All the attribution for this patch should go to Snom Technology AG*

### CODE REVIEW

File: src/call.cpp [hunk 1]
```diff
@@ -3510,6 +3510,10 @@ call::T_ActionResult call::executeAction(char * msg, message *curmsg)
         } else if (currentAction->getActionType() == CAction::E_AT_PAUSE_RESTORE) {
             double operand = get_rhs(currentAction);
             paused_until = (int)operand;
+#ifdef RTP_STREAM
+        } else if (currentAction->getActionType() == CAction::E_AT_RTP_ECHO) {
+            rtp_echo_state = (currentAction->getDoubleValue() != 0 ? true : false);
+#endif
```
Comment (wdoekes) [inline src/call.cpp]: There is an RTP_STREAM ifdef below on line 3693, why not put it in there.

And `rtp_echo_state = (currentAction->getDoubleValue() != 0)` is explicit enough for a boolean assignment.

File: src/scenario.cpp [hunk 2]
```diff
@@ -1481,6 +1481,11 @@ void scenario::parseAction(CActions *actions)
         } else if(!strcmp(actionElem, "pauserestore")) {
             tmpAction->setActionType(CAction::E_AT_PAUSE_RESTORE);
             handle_rhs(tmpAction, "pauserestore");
+#ifdef RTP_STREAM
+        } else if(!strcmp(actionElem, "rtp_echo")) {
+            tmpAction->setActionType(CAction::E_AT_RTP_ECHO);
+            handle_rhs(tmpAction, "rtp_echo");
+#endif
```
Comment (wdoekes) [inline src/scenario.cpp]: There is an #ifdef already at 1634.. please move it closer.

File: src/sipp.cpp [hunk 3]
```diff
@@ -603,6 +603,11 @@ static void rtp_echo_thread(void* param)
                     errno);
             return;
         }
+#ifdef RTP_STREAM
+        if (rtp_echo_state == false) {
```
Comment (wdoekes) [inline src/sipp.cpp]: `!rtp_echo_state`

**Comment by pbertera** [comment]
Thanks @wdoekes for the review and feedback. This should be better now.

File: src/scenario.cpp [hunk 4]
```diff
@@ -1647,6 +1647,13 @@ void scenario::parseAction(CActions *actions)
             } else {
                 ERROR("illegal <exec> in the scenario\n");
             }
+        } else if(!strcmp(actionElem, "rtp_echo")) {
+#ifdef RTP_STREAM
+            tmpAction->setActionType(CAction::E_AT_RTP_ECHO);
+            handle_rhs(tmpAction, "rtp_echo");
+#else
+                ERROR("Scenario specifies a rtp_echo action, but this version of SIPp does not have RTP stream support");
```
Comment (wdoekes) [inline src/scenario.cpp]: Indentation nitpick.

File: regress/github-#0259/run [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+test $status -ne 0 && fail "SIPp UAC job failed"
+
+if test "$port" = "$uac_media_port"; then
+    ok
+elif test -n "$port"; then
+    fail "got RTP from wrong source port $port"
+else
+    fail "got no RTP at all"
+fi
```
Comment (wdoekes) [inline regress/github-#0259/run]: Lines 21..27 are NO-OPS since $port and $uac_media_port are blank.

Replace with a single `ok`.

**Comment by wdoekes** [comment]
Looks good.

I wonder why this just happened in one travis check:
```
checking for unistd.h... yes
checking openssl/md5.h usability... no
checking openssl/md5.h presence... no
checking for openssl/md5.h... no
configure: error: <openssl/md5.h> header missing
The command "./configure $CONFOPTS" failed and exited with 1 during .
```

**Comment by wdoekes** [comment]
Something has changed with the OSX travis build. Investigating... I hope it's fixed with: https://travis-ci.org/SIPp/sipp/builds/165197224

**Comment by wdoekes** [comment]
Ok. Should be fixed by 7412b02bf. Can you rebase against new master?

**Comment by pbertera** [comment]
Thanks, now travis build looks good

**Pull request merged**